cli/tools/compile.rs

```diff
@@ -378,27 +379,24 @@ fn get_module_roots_and_include_files(
 
   let mut searched_paths = HashSet::new();
   let mut module_roots = Vec::new();
-  let mut include_files = Vec::new();
+  let mut include_paths = Vec::new();
   module_roots.push(entrypoint.clone());
   for side_module in &compile_flags.include {
     let url = resolve_url_or_path(side_module, initial_cwd)?;
     if is_module_graph_module(&url) {
       module_roots.push(url.clone());
-      if url.scheme() == "file" {
-        include_files.push(url);
-      }
     } else {
       analyze_path(&url, &mut searched_paths, |file_url| {
-        if file_url != *output_url {
-          include_files.push(file_url.clone());
-          if is_module_graph_module(&file_url) {
-            module_roots.push(file_url);
-          }
+        if is_module_graph_module(&file_url) {
+          module_roots.push(file_url);
         }
```

Comment:
This is not as efficient because we now potentially traverse directories twice, but it is more correct now.